### ISSUE TITLE
Fix build errors with MEMORY_HOTPLUG=y && FLATMEM

### ISSUE THREAD

**Issue by mpe** [linuxppc/linux#56]
In most configurations we force`SPARSEMEM`, but it is possible to enable `FLATMEM`. In that case if you also enable `MEMORY_HOTPLUG` the builds fails:

```
include/linux/memory.h:23:43: error: 'SECTION_SIZE_BITS' undeclared (first use in this function)
arch/powerpc/platforms/pseries/hotplug-memory.c:125:15: error: implicit declaration of function 'pfn_to_section_nr' [-Werror=implicit-function-declaration]
mm/memory_hotplug.c:1236:20: error: 'PAGE_SECTION_MASK' undeclared (first use in this function)
```

And so on.

We should either require `SPARSEMEM` always (for pseries/powernv), or make `MEMORY_HOTPLUG` depend on `SPARSEMEM` somehow.

More details here: http://kisskb.ellerman.id.au/kisskb/buildresult/12664645/

**Comment by deece** [comment]
We currently have:

config ARCH_FLATMEM_ENABLE
        def_bool y
        depends on (PPC64 && !NUMA) || PPC32

Both pseries & powernv force PPC64, but only powernv forces NUMA.

@mpe Should pseries alse force NUMA?

**Comment by mpe** [comment]
powernv doesn't force NUMA=y, it is just the default in powernv_defconfig.

But I think this is now fixed by 577ec789a79e `powerpc/cell: Drop select of MEMORY_HOTPLUG`.

That was the root cause in the build error linked above.

And looking at `mm/Kconfig` we have:
```
config MEMORY_HOTPLUG
	bool "Allow for memory hot-add"
	depends on SPARSEMEM || X86_64_ACPI_NUMA
```

So `MEMORY_HOTPLUG` does depend on `SPARSEMEM` (and always did), and now that we've removed the `select` of `MEMORY_HOTPLUG` from the spu code there's no way to enable it unless `SPARSEMEM` is enabled.

I'll close this and we can reopen if we see it again.
